libsupport/src/Logging.cpp

```diff
@@ -0,0 +1,43 @@
+#include "galois/Logging.h"
+
+#include <iostream>
+#include <mutex>
+
+#include "galois/GetEnv.h"
+
+namespace {
+
+void PrintString(bool error, bool flush, const std::string& prefix,
+                 const std::string& s) {
+  static std::mutex lock;
+  std::lock_guard<std::mutex> lg(lock);
+
+  std::ostream& o = error ? std::cerr : std::cout;
+  if (!prefix.empty()) {
+    o << prefix << ": ";
+  }
+  o << s << "\n";
+  if (flush) {
+    o.flush();
+  }
+}
+
+} // end unnamed namespace
+
+void galois::internal::LogString(galois::LogLevel level, const std::string& s) {
```

Comment:
+1 to configuring the output stream here. That'll make OS compatibility stuff easier to handle later.